Browser_IDE/editorMain.js

```diff
@@ -488,10 +488,26 @@ async function FSdownloadFile(filename, mime) {
 }
 
 
-async function downloadProject(){
-    downloadFileGeneric(await projectToZip(), "project.zip");
+async function downloadProject(projectName){
+    let projectName = (await appStorage.access((s) => s.getProject(storedProject.projectID))).name;
+    downloadFileGeneric(await projectToZip(), projectName + ".zip");
```

Comment:
This is close, but right now doesn't run since `projectName` is defined twice. Just make the signature `async function downloadProject()` and it'll work.